drivers/vfs/driver.go

```diff
@@ -244,6 +248,37 @@ func (d *Driver) Remove(id string) error {
 	return system.EnsureRemoveAll(d.dir(id))
 }
 
+func (d *Driver) GetTempDirRootDirs() []string {
+	tempDirs := []string{filepath.Join(d.home, tempDirName)}
+	// Only include imageStore if it's different from home
+	// Writable layers can only be in d.home or d.imageStore, not in additionalHomes (which are read-only)
+	if d.imageStore != "" && d.imageStore != d.home {
```

Comment:
The `d.imageStore != d.home` condition seems unnecessary, and inconsistent with other users of `d.imageStore`.

 `loadStoreOptionsFromConfFile` already refuses options of ~that kind; but, also, `d.imageStore == d.home` is not _exactly_ the right condition _anyway_, because the layouts are different (the `d.String()` part).

Also in `DeferredRemove`.